### PR TITLE
Fix regression for short lived processes

### PR DESCRIPTION
**Pull request by r1viollet**
# What does this PR do?

This is a quick fix that consists in refreshing the path for files to the last know location. A more long term fix would consist in finding the absolute path across containers.

# Motivation

Reliability environment caught this regression.

### CODE REVIEW

File: include/device_utils.hpp [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+#  undef custom_makedev
+#  undef custom_major
+#  undef custom_minor
+
+#  if defined(__linux__) || defined(__GLIBC__)
+/* Linux, Android, and other systems using GNU C library */
+#    include <sys/sysmacros.h>
+#    define custom_makedev(dmajor, dminor) makedev(dmajor, dminor)
+#    define custom_major(devnum) major(devnum)
```
Comment (r1viollet) [inline include/device_utils.hpp:15]: This is not super useful for now, though we will need the API for the more long term fix.

File: test/dwfl_module-ut.cc [hunk 2]
```diff
@@ -97,4 +101,78 @@ TEST(DwflModule, inconsistency_test) {
...
+  waitpid(child_pid, nullptr, 0);
+
+  pid_t last_child_pid = fork();
+  if (last_child_pid == 0) {
+    // First child process
+    sleep(2);
+    exit(0);
+  }
+
+  // Parse the first pid
```
Comment (sanchda) [inline test/dwfl_module-ut.cc:153]: s/first/second/ ?

**Review by sanchda — APPROVED** [review]
LGTM

**Review by sanchda — DISMISSED** [review]
LGTM

**Review by nsavoire — APPROVED** [review]

File: test/dwfl_module-ut.cc [hunk 3]
```diff
@@ -97,4 +101,78 @@ TEST(DwflModule, inconsistency_test) {
...
+      ASSERT_TRUE(ddprof_mod->_mod);
+    }
+  }
+  // Wait for the first PID to die
+  waitpid(child_pid, nullptr, 0);
+
+  pid_t last_child_pid = fork();
+  if (last_child_pid == 0) {
+    // First child process
+    sleep(2);
```
[the remark this answers is not in the document]
Comment (r1viollet) [reply test/dwfl_module-ut.cc]: Fat fingers :smile:

**Pull request merged**